saleor/dashboard/product/forms.py

```diff
@@ -123,19 +124,34 @@ def clean(self):
                 'and its variant.')
             self.add_error('variant_attributes', msg)
 
-        if self.instance.pk:
-            variants_changed = (
-                self.fields['has_variants'].initial != has_variants)
-            if variants_changed:
-                query = self.instance.products.all()
-                query = query.annotate(variants_counter=Count('variants'))
-                query = query.filter(variants_counter__gt=1)
-                if query.exists():
-                    msg = pgettext_lazy(
-                        'Product type form error',
-                        'Some products of this type have more than '
-                        'one variant.')
-                    self.add_error('has_variants', msg)
+        if not self.instance.pk:
+            return data
+
+        variants_changed = (
```

Comment:
Django's [has_changed](https://docs.djangoproject.com/en/2.0/ref/forms/fields/#checking-if-the-field-data-has-changed) wouldn't help here?

Reply:
Unfortunately not